ICAFIX/ReApplyFixProcessingWrapper.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+set -eu
+
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+g_matlab_default_mode=1
+
+#mandatory (mrfix name must be specified if applicable, so including it here despite being mechanically optional)
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject' 'Subject' '100206' "one subject ID"
+opts_AddMandatory '--fmri-names' 'fMRINames' 'rfMRI_REST1_LR@rfMRI_REST1_RL...' "list of fmri run names separated by @s" #Needs to be the single fMRI run names only (for DVARS and GS code) for MR+FIX, is also the SR+FIX input names
+opts_AddOptional '--mrfix-concat-name' 'MRFixConcatName' 'rfMRI_REST' "if multi-run FIX was used, you must specify the concat name with this option" ""
+opts_AddMandatory '--fix-high-pass' 'HighPass' 'integer' 'the high pass value that was used when running FIX' '--melodic-high-pass'
+opts_AddMandatory '--surf-reg-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files"
+opts_AddConfigMandatory '--low-res' 'LowResMesh' 'LowResMesh' 'meshnum' "mesh resolution, like '32' for 32k_fs_LR"
+opts_AddOptional '--motion-regression' 'MotionReg' 'MotionReg' " whether to apply motion regression" "FALSE"
+opts_AddOptional '--delete-intermediates' 'DeleteIntermediates' 'DeleteIntermediates' "whether to delete intermediate files" "FALSE"
+opts_AddOptional '--matlab-run-mode' 'MatlabMode' '0, 1, or 2' "defaults to $g_matlab_default_mode
+
+0 = compiled MATLAB
+1 = interpreted MATLAB
+2 = Octave" "$g_matlab_default_mode"
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+if [ ${RegName} = "NONE" ] ; then
+    RegNameString=""
+else
+    RegNameString="_${RegName}"
+fi
+
+#processing code goes here
+if [ -z ${MRFixConcatName} ] ; then
+    # if single run
+    #Move old files to backup
+    echo "" > /dev/null
+    for fMRIName in "${fMRINamesArray[@]}"; do
+        mv -n ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_Atlas${RegNameString}_hp${HighPass}_clean.dtseries.nii ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_Atlas${RegNameString}_hp${HighPass}_clean_bak.dtseries.nii
+        mv -n ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_Atlas${RegNameString}_hp${HighPass}_clean_vn.dscalar.nii ${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_Atlas${RegNameString}_hp${HighPass}_clean_vn_bak.dscalar.nii
```

Comment:
Once we have merged Yash's refactored branch, I strongly suggest we add an option to `ReApply...` to change the hardcoded suffix, rather than this brittle "move all the outputs we currently make, rerun, and rename everything again" script.

Reply:
The same functionality has been taken care of in the refactor branch?

Reply:
The refactored branch change the argument parsing style for ReApply.  Now that it is merged, we can add the option in the new style (rather than creating it in the old style and having a merge conflict).

Reply:
To be clear, we should rebase your branch onto master (and resolve the conflict) before adding that option to ReApply.

Reply:
Okay, I see. You are suggesting to add this functionality to `ReApply`, instead of using the current wrapper script. Can you rebase this branch and then let me just run `git pull` to get it rebased?

Reply:
Oh, I see the conflict...the reclassify filenames used to be hardcoded, which prevented the use of reclean without overwriting them, that was an unfortunate initial design.  I'll see about adding them as defaults in the new style, but you'll need to test it.

Reply:
Should be ready now.

Reply:
I will revise the `ReApply` tomorrow.

Reply:
You just did a merge commit with your old branch before I rebased it...I'll see about trying to clean it up.

Reply:
I believe I rescued the edits that got baked into your merge commit.  Before you pull again, please do the following sequence:
```
git stash
git branch savecurrent
git remote update
git checkout --detach
git branch -f master origin/master
git checkout master
```
, assuming cyang31 is your origin.  If "git stash" did anything (it saves uncommitted edits), you can do "git stash apply" to restore them afterwards.  If I didn't get all your edits, the "savecurrent" branch will still have them, and we can figure out how to retrieve them (but git diff told me my rescued version was the same).

In the future, please use `--ff-only` every time you `git pull`, so it won't make an automatic merge commit.